File: python/defaults/component/routing.py
#
# Prudence Routing
#

from java.io import File

# Hosts

include_or_default('component/hosts')

# Applications

applications = component.context.attributes['applications'] = []
application_dirs = File('applications').listFiles()
for application_dir in application_dirs:
    if application_dir.isDirectory():
        application_name = application_dir.name
        application_internal_name = application_dir.name;
        application_logger_name = application_dir.name
        application_base_path = application_dir.path
        application_default_url = '/' + application_dir.name + '/'
        include_or_default(application_base_path, 'defaults/application')
        applications.append(application)

if len(applications) == 0:
    print 'No applications found. Exiting.'
    System.exit(0)

File: javascript/defaults/component/routing.js
//
// Prudence Routing
//

importClass(
	java.io.File,
	java.util.ArrayList);

// Hosts

includeOrDefault('component/hosts');

// Applications

var applications = new ArrayList();
component.context.attributes.put('applications', applications);
var applicationDirs = new File('applications').listFiles();
for(var i in applicationDirs) {
	var applicationDir = applicationDirs[i]; 
	if(applicationDir.isDirectory()) {
		var applicationName = applicationDir.name;
		var applicationInternalName = applicationDir.name;
		var applicationLoggerName = applicationDir.name;
		var applicationBasePath = applicationDir.path;
		var applicationDefaultURL = '/' + applicationDir.name + '/';
		includeOrDefault(applicationBasePath, 'defaults/application');
		applications.add(application);
	}
}

if(applications.length == 0) {
	print('No applications found. Exiting.\n');
	System.exit(0);
}
